### PR TITLE
MDS-317 : Updated configs based on new data items for endpoints.

### PR DESCRIPTION
**Pull request by grab-a-byte**

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -163,47 +163,44 @@ versioning {
...
-        E = "employer/employerAddress/line3"
-        F = "employer/employerDistrictNumber"
-        G = "employer/employerSchemeReference"
-        H = "employmentStartDate"
-        I = "employmentLeavingDate"
-        J = "employmentPayFrequency"
-        K = "employeeAddress"
-        L = "payrollId"
+    details {
+        endpoint = "/individuals/details/"
```
Comment (ajchilts) [inline conf/application.conf]: I don't think this needs to be a route. It just needs to match the endpoint name... 

I.e. https://github.com/hmrc/individuals-details-api/blob/master/app/uk/gov/hmrc/individualsdetailsapi/controllers/ContactDetailsController.scala#L35

File: conf/application.conf [hunk 2]
```diff
@@ -163,47 +163,44 @@ versioning {
...
-        D = "employer/employerAddress/line2"
-        E = "employer/employerAddress/line3"
-        F = "employer/employerDistrictNumber"
-        G = "employer/employerSchemeReference"
-        H = "employmentStartDate"
-        I = "employmentLeavingDate"
-        J = "employmentPayFrequency"
-        K = "employeeAddress"
-        L = "payrollId"
+    details {
```
Comment (ajchilts) [inline conf/application.conf]: I don't think this config is correct? 

Looking at the data matrix we need two endpoints in the individual details api.

```
/details
    /contact-details
    /addresses 
```

I think we need to add a new unimplemented endpoint and align to the design as outlines in the data matrix.

**Review by ajchilts — CHANGES_REQUESTED** [review]
Fantastic effort however; I think we need to alter the solution to match the data matrix as outlined in my review comments.

File: conf/application.conf [hunk 3]
```diff
@@ -163,47 +163,44 @@ versioning {
...
+    }
+    "read:individuals-details-lsani-c3" {
+       fields = ["A", "B", "C", "D", "E", "F", "I", "J"]
+    }
+    "read:individuals-details-nictsejo-c4" {
+       fields = ["A", "B", "C", "D", "E", "F", "I", "J"]
     }
   }
 
   endpoints {
```
Comment (ajchilts) [inline conf/application.conf:189]: Also this endpoint config looks different to that in the test config used bu the code for testing the handler pattern. 

```
    endpoints = List(
      EndpointConfig(
        name = mockEndpoint1,
        link = "/a/b/c?matchId=<matchId>{&fromDate,toDate}",
        fields = Map(
          "A" -> "payments",
          "B" -> "employer/employerName=\"foo bar\"",
          "C" -> "employer/employerAddress/line1",
          "D" -> "employer/employerAddress/line2",
          "E" -> "employer/employerAddress/line3",
          "F" -> "employer/employerDistrictNumber",
          "G" -> "employer/employerSchemeReference",
          "H" -> "employmentStartDate",
          "I" -> "employmentLeavingDate",
          "J" -> "employmentPayFrequency",
          "K" -> "employeeAddress",
          "L" -> "payrollId"
        )
      ),
```

I think we will need to align to this name, link, fields 

Name is the string name of the endpoint used by the endpoint to fetch the scopes. 

Link is the full url for HAL 

Fields are the fields. 

@owen-opencast  can help confirm this?

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: conf/application.conf (hunk 1 above)
Comment (ajchilts) [inline conf/application.conf]: Update to correct HAL link and ensure endpoint keys match those in the controllers `getEndPointScopes` invocation.

**Review by ajchilts — CHANGES_REQUESTED** [review]

**Review by owen-gilfellon — APPROVED** [review]
Looks good, just update the HATEOAS links

File: conf/application.conf (hunk 3 above)
Comment (grab-a-byte) [reply conf/application.conf:189]: parsing step in config handler makes this diference

File: test/component/uk/gov/hmrc/individualsdetailsapi/controllers/SandboxAddressesControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package component.uk.gov.hmrc.individualsdetailsapi.controllers
+
+import component.uk.gov.hmrc.individualsdetailsapi.stubs.{AuthStub, BaseSpec}
+import play.api.test.Helpers._
+import scalaj.http.Http
+
+class SandboxAddressesControllerSpec extends BaseSpec {
+  val rootScope = "read:individuals-details-contact-details"
```
Comment (ajchilts) [inline test/component/uk/gov/hmrc/individualsdetailsapi/controllers/SandboxAddressesControllerSpec.scala]: Should this not match the live controller for consistency? 

List("read:individuals-details-hmcts-c4", "read:individuals-details-laa-c4")

Comment (ajchilts) [reply test/component/uk/gov/hmrc/individualsdetailsapi/controllers/SandboxAddressesControllerSpec.scala]: Also do we need to update the sandbox contact details details controller?

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: conf/application.conf [hunk 5]
```diff
@@ -163,47 +163,49 @@ versioning {
...
-        F = "employer/employerDistrictNumber"
-        G = "employer/employerSchemeReference"
-        H = "employmentStartDate"
-        I = "employmentLeavingDate"
-        J = "employmentPayFrequency"
-        K = "employeeAddress"
-        L = "payrollId"
-      }
+    addresses {
+        endpoint = "/individuals/details/addresses?matchId=$randomMatchId"
```
Comment (owen-gilfellon) [inline conf/application.conf]: Can we change the HATEOAS links to the form:
/individuals/details/addresses?matchId=<matchId> - we'll then search and replace <matchId> when returning the links

**Review by owen-gilfellon — CHANGES_REQUESTED** [review]

File: conf/application.conf (hunk 5 above)
Comment (grab-a-byte) [reply conf/application.conf]: I think search and replace $randomMatchId or something similar is better as some of the endpoints have "-" in them and as such having something explicit like this makes more sense to me

**Review by ajchilts — APPROVED** [review]
Great effort 👍

**Pull request merged**